wurst/util/StringUtils.wurst

```diff
@@ -157,6 +126,101 @@ public function string.reduce(string reduced) returns string
 public function string.reduce() returns string
 	return this.reduce(" ")
 
+public interface ForStringSplitCallback
+	function callback(string str)
+
+/** Custom Function which can split strings and runs callback for each string.
+	It can run even with no delimiter present (with single element).
+	Example usage:
+	"H001,H002".forEachIn(",") ->
+		doSomething()  */
+public function string.forEachIn(string delim, ForStringSplitCallback cb)
+	if this.length() > 0
+		int i = 0
+		int pos = this.indexOf(delim, i)
+
+		if pos == -1
+			cb.callback(this.substring(i, this.length()))
+
+		while pos != -1
+			cb.callback(this.substring(i, pos))
+			pos += 1
+			i = pos
+			pos = this.indexOf(delim, pos)
+			if pos == -1
+				cb.callback(this.substring(i, this.length()))
+
+/** Creates list of IDs converted from Int to their String value delimited by comma (using toRawCode).*/
+public function createDelimitedString(vararg int ids) returns string
+	string s = ""
+	var paramCount = 0
+	for id in ids
+		paramCount++
+		s += id.toRawCode()
+		s += ","
+	if s != ""
+		s = s.substring(0, s.length()-1)
+	if paramCount > WC3LIMIT_FUNCTIONS_PARAMETER_COUNT
+		error(WC3LIMIT_FUNCTIONS_PARAMETER_COUNT_ERROR)
+	return s
+
+/** Creates one long String from all Strings delimited by comma.*/
+public function createDelimitedString(vararg string ids) returns string
+	string s = ""
+	var paramCount = 0
+	for id in ids
+		paramCount++
+		if id != ""
+			s += id
+			s += ","
+	if s != ""
+		s = s.substring(0, s.length()-1)
+	if paramCount > WC3LIMIT_FUNCTIONS_PARAMETER_COUNT
+		error(WC3LIMIT_FUNCTIONS_PARAMETER_COUNT_ERROR)
+	return s
+
+/** Checks whether a string is representing a digit or not.*/
+public function string.isDigit() returns bool
+	bool isDigit = false
+	for int i = 0 to this.length()-1
+		if this.charAt(i) == "0" or this.charAt(i) == "1" or this.charAt(i) == "2" or this.charAt(i) == "3" or this.charAt(i) == "4"
+		or this.charAt(i) == "5" or this.charAt(i) == "6" or this.charAt(i) == "7" or this.charAt(i) == "8" or this.charAt(i) == "9"
+			isDigit = true
+		else
+			isDigit = false
+			return isDigit
+	return isDigit
+
+/** Checks if this string contains all characters in given string.
+	Can be useful for player commands as you can capture entered strings with typos.
+	If this string has different length than given string, it immediately returns false.
+	Example: "-etrne".containsAllAndOnly("-enter") == true */
+public function string.containsAllAndOnly(string chars) returns bool
+	bool containsAll = true
+	if this.length() == chars.length()
+		for char in chars
+			if not this.contains(char)
+				containsAll = false
+				break
+	else
+		containsAll = false
+	return containsAll
+
+/** Checks whether a string is valid in terms of max WC3 string length for various usages.*/
+public function string.isValidLength() returns bool
```

Comment:
Specify `various` usages, or rather, rename to something properly, like `isValidDescription`. This is too generic.